website/i18n/fr/docusaurus-plugin-content-docs/current/concepts/combining_providers.mdx

```diff
@@ -0,0 +1,278 @@
+
+final charactersProvider = FutureProvider<List<Character>>((ref) async {
+  final search = ref.watch(searchProvider);
+  final configs = await ref.watch(configsProvider.last);
+  final response = await dio.get('${configs.host}/characters?search=$search');
+
+  return response.data.map((json) => Character.fromJson(json)).toList();
+});
+```
+
+Ce code récupère une liste de caractères à partir du service, 
+et récupère à nouveau automatiquement la liste, chaque fois que les 
+configurations changent ou que la requête de recherche change.
+:::
+
+### Peut-on lire un provider sans l'écouter ?
+
+Parfois, on souhaite lire le contenu d'un provider, mais sans recréer la valeur 
+exposée lorsque la valeur obtenue change.
+
+Un exemple serait un `Repository`, qui lit depuis un autre provider le jeton de l'utilisateur
+pour l'authentification.
+Nous pourrions utiliser [watch] et créer un nouveau `Repository` chaque fois que le jeton de l'utilisateur change, 
+mais il y a peu ou pas d'utilité à faire cela.
+
+Dans cette situation, nous pouvons utiliser [read], qui est similaire à [watch], mais qui n'obligera pas le provider 
+à recréer sa valeur exposée lorsque la valeur obtenue change.
+
+Dans ce cas, une pratique commune est de faire passer `ref.read` à l'objet créé. 
+Cet objet sera alors capable de lire les providers quand il le souhaite.
+
+```dart
+final userTokenProvider = StateProvider<String>((ref) => null);
+
+final repositoryProvider = Provider((ref) => Repository(ref.read));
+
+class Repository {
+  Repository(this.read);
+
+  /// La fonction `ref.read` 
+  final Reader read;
+
+  Future<Catalog> fetchCatalog() async {
+    String token = read(userTokenProvider).state;
+
+    final response = await dio.get('/path', queryParameters: {
+      'token': token,
+    });
+
+    return Catalog.fromJson(response.data);
+  }
+}
+```
+
+:::note
+Vous pouvez aussi passer le `ref` au lieu de `ref.read` à votre objet :
+
+```dart
+final repositoryProvider = Provider((ref) => Repository(ref));
+
+class Repository {
+  Repository(this.ref);
+
+  final Ref ref;
+}
+```
+
+La seule différence que le passage de `ref.read` apporte est un code 
+un peu moins verbeux et la garantie que notre objet n'utilise jamais `ref.watch`.
+:::
+
+:::danger **NE PAS** appelez [read] à l'intérieur du provider.
+
+```dart
+final myProvider = Provider((ref) {
+  // Mauvaise pratique d'appeler `read` ici.
+  final value = ref.read(anotherProvider);
+});
+```
+
+Si vous avez utilisé [read] pour tenter d'éviter les reconstructions non souhaitées de votre objet, reportez-vous à 
+[Mon provider fait des mises à jour trop souvent, que puis-je faire ?](#my-provider-updates-too-often-what-can-i-do)
+Mauvaise pratique pour appeler `read` ici
+:::
+
+### Comment tester un objet qui reçoit [read] comme paramètre de son constructeur ?
+
+Si vous utilisez le modèle décrit dans [Puis-je lire un provider sans l'écouter ?](#can-i-read-a-provider-without-listening-to-it), 
```

Comment:
Those links need to be updated to use the French description

Reply:
There are a few of those, so they should all be edited :)

Reply:
I'll go through all of them again, no problem 👍